frontend/public/components/resource-dropdown.tsx

```diff
@@ -1,123 +1,43 @@
-
-To avoid circular imports, the keys in this list are manually duplicated in ./resource-pages.tsx !
-
-------------------------------------------------------------------------
-*/
-const resources = [
-  'Alertmanagers',
-  'ClusterServiceVersion-v1s',
-  'Clusters',
-  'ConfigMaps',
-  'CronJobs',
-  'CustomResourceDefinitions',
-  'DaemonSets',
-  'Default',
-  'Deployments',
-  'EtcdClusters',
-  'HorizontalPodAutoscalers',
-  'Ingresses',
-  'Jobs',
-  'Namespaces',
-  'NetworkPolicies',
-  'Nodes',
-  'PersistentVolumeClaims',
-  'PersistentVolumes',
-  'PodVulns',
-  'Pods',
-  'Prometheuses',
-  'ReplicaSets',
-  'ReplicationControllers',
-  'ReportGenerationQuerys',
-  'Reports',
-  'ResourceQuotas',
-  'RoleBindings',
-  'Roles',
-  'Secrets',
-  'ServiceAccounts',
-  'ServiceMonitors',
-  'Services',
-  'StatefulSets',
-  'StorageClasses',
-];
-
-const openshiftResources = [
-  'BuildConfigs',
-  'Builds',
-  'DeploymentConfigs',
-  'ImageStreamTags',
-  'ImageStreams',
-  'Projects',
-  'Routes',
-];
-
-const DropdownItem = ({kind}) => {
-  const [modelRef, kindObj] = allModels().findEntry((v) => v.kind === kind);
-  return <span>
-    <span className="co-type-selector__icon-wrapper">
-      <ResourceIcon kind={modelRef} />
-    </span>
-    {kindObj.labelPlural}
-  </span>;
-};
 
-const ResourceListDropdown_: React.StatelessComponent<ResourceListDropdownProps> = props => {
-  const { selected, onChange, allkinds, openshiftFlag, showAll, className } = props;
-  if (flagPending(openshiftFlag)) {
-    return null;
-  }
-
-  const items: {[s: string]: JSX.Element} = {};
-  const kinds = {};
-  _.each(allModels().toJS(), (ko: K8sKind) => kinds[ko.labelPlural.replace(/ /g, '')] = ko.kind);
-
-  if (showAll) {
-    items.all = <span>
-      <span className="co-type-selector__icon-wrapper">
-        <ResourceIcon kind="All" />
-      </span>All Types</span>;
-  }
-
-  const allResources = openshiftFlag ? _.concat(resources, openshiftResources) : resources;
-  allResources.filter(k => k !== ClusterServiceVersionModel.labelPlural)
-    .sort()
-    .forEach(k => {
-      const kind: string = kinds[k];
-      if (!kind) {
-        return;
-      }
-      const model = allkinds[kind];
-      if (model && model.crd && ![EtcdClusterModel, PrometheusModel, ServiceMonitorModel, AlertmanagerModel].some(m => m.kind === k)) {
-        return;
-      }
-      items[kind] = <DropdownItem kind={kind} />;
-    });
+import { Dropdown, ResourceIcon } from './utils';
+import { K8sKind, K8sResourceKindReference, referenceForModel } from '../module/k8s';
+
+const DropdownItem: React.SFC<DropdownItemProps> = ({model}) => <React.Fragment>
+  <span className="co-type-selector__icon-wrapper">
+    <ResourceIcon kind={model.kind} />
+  </span>
+  {model.kind}
+</React.Fragment>;
+
+const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = props => {
+  const { selected, onChange, allModels, showAll, className } = props;
+
+  const items = (allModels
```

Comment:
Now we show all registered API resources, no more whitelists.

Reply:
We'd want to remove duplicate resources from the same API group, but different API versions:

![screen shot 2018-07-18 at 4 29 08 pm](https://user-images.githubusercontent.com/1167259/42906026-cabbce84-8aa7-11e8-9290-b90575e00eb6.png)

Reply:
@spadgett How do we know which API version to keep? We need to allow searching old API versions.

Reply:
API groups have a preferred version:

```json
    {
      "name": "apps",
      "versions": [
        {
          "groupVersion": "apps/v1",
          "version": "v1"
        },
        {
          "groupVersion": "apps/v1beta2",
          "version": "v1beta2"
        },
        {
          "groupVersion": "apps/v1beta1",
          "version": "v1beta1"
        }
      ],
      "preferredVersion": {
        "groupVersion": "apps/v1",
        "version": "v1"
      }
    }
```

> We need to allow searching old API versions.

Not necessarily saying you're wrong, but what is the use case? Maybe we add a toggle to show old versions?

We at least need to make sure there's a way to tell the versions apart in the dropdown.